### PR TITLE
Integrated Service Discovery + GitHub Linking

### PR DESCRIPTION
**Pull request by pingrishabh**

### CODE REVIEW

File: server/api/routes/service.ts
Comment (utkarsh-dixit) [inline server/api/routes/service.ts]: I think it should be email here, the User Model doesn't have a `mobile` column as of now.

File: server/api/routes/service.ts
Comment (utkarsh-dixit) [inline server/api/routes/service.ts:14]: Lets create a middleware for this because jwt verification would be used in other routes too.

File: server/api/services/service.ts
Comment (utkarsh-dixit) [inline server/api/services/service.ts]: `ServiceService` name sounds ambiguous, maybe something like `ServiceProvidersService`?

File: server/api/controllers/service.ts
Comment (utkarsh-dixit) [inline server/api/controllers/service.ts]: ServiceService name sounds ambiguous, maybe something like ServiceProvidersService?

**Review by utkarsh-dixit — CHANGES_REQUESTED** [review]
Let's merge this after the above comments are resolved.

 @Cheesetouched Amazing work!! the code was very well organised :fire:

File: server/api/routes/service.ts
Comment (pingrishabh) [reply server/api/routes/service.ts]: Oh! The service discovery endpoint was working even after this error, because it picked the first user as a fallback, and I was the first user in the db. What a coincidence. 😆 

Anyway, it was super important to catch this. Good eye! @utkarsh-dixit

File: server/api/services/service.ts
Comment (pingrishabh) [reply server/api/services/service.ts]: I agree. Going for ServiceProviderService.

File: server/api/controllers/service.ts
Comment (pingrishabh) [reply server/api/controllers/service.ts]: Done!

**Review by flozender — APPROVED** [review]

File: server/.env.example
Comment (flozender) [inline server/.env.example]: Is this file supposed to be committed?

File: server/api/controllers/service.ts
Comment (flozender) [inline server/api/controllers/service.ts:71]: Why do we need the ```redirect_url``` to get an access token? Is it required?

Comment (flozender) [reply server/api/controllers/service.ts:71]: Ok, sorry, I just found out that we do need it. Do you want me to send it in through the frontend or will you guys populate env variables? I think it's better I send the ```redirect_url``` to the backend.

File: server/.env.example
Comment (utkarsh-dixit) [inline server/.env.example]: Add a new line at the end

Edit: This file needs to be removed, and should not be committed in the repository.

File: server/api/routes/service.ts
Comment (utkarsh-dixit) [reply server/api/routes/service.ts]: I can still see `mobile` here, re-opening the conversation again

**Review by utkarsh-dixit — CHANGES_REQUESTED** [review]

File: server/.env.example
Comment (utkarsh-dixit) [reply server/.env.example]: I think this got committed by mistake. @anje123 Can you remove it from the PR?

Comment (anje123) [reply server/.env.example]: ok bro

File: server/api/controllers/service.ts
Comment (anje123) [reply server/api/controllers/service.ts:71]: sure it better through the frontend

Comment (flozender) [reply server/api/controllers/service.ts:71]: Cool, pushing it in my next PR EDIT: merged to master

**Review by utkarsh-dixit — CHANGES_REQUESTED** [review]
@Cheesetouched still needs to push his changes, after which we can go ahead and merge this.

File: server/api/routes/service.ts
Comment (utkarsh-dixit) [reply server/api/routes/service.ts:14]: I will create a middleware for this in other PR, resolve it now

**Pull request merged**
